app/server.go

```diff
@@ -531,7 +531,7 @@ func NewServer(options ...Option) (*Server, error) {
 	} else {
 		nErr := backend.TestConnection()
 		if nErr != nil {
-			if errors.Is(nErr, filestore.ErrNoS3Bucket) {
+			if _, ok := nErr.(*filestore.S3FileBackendNoBucketError); ok {
```

Comment:
Can we not use `errors.Is` with `filestore.S3FileBackendNoBucketError`?

Reply:
Using `errors.Is` would require `filestore.S3FileBackendNoBucketError` to be an actual variable, not a type. I decided to use a type so that I could easily use a type switch in `connectionTestErrorToAppError` in `app/file.go`.